pkg/kubelet/dockertools/docker.go

```diff
@@ -1049,3 +1049,21 @@ func RunContainer(client DockerInterface, container *api.Container, pod *api.Pod
 	}
 	return dockerContainer.ID, nil
 }
+
+// KillContainer kills a docker container with the given containerID.
+// TODO(yifan): To use strong type for the containerID.
+func KillContainer(client DockerInterface, containerID string,
```

Comment:
If the runtime does do it implicitly I agree that it should set the readiness itself (or provide some information about the restarted containers). I don't want the Kubelet to do something different depending on the runtime so the approach you have today is preferred.

I will wait for your rethinking though :)